core/trino-main/src/main/java/io/trino/sql/planner/plan/TableWriterNode.java

```diff
@@ -84,6 +86,7 @@ public TableWriterNode(
         this.columnNames = ImmutableList.copyOf(columnNames);
         this.notNullColumnSymbols = ImmutableSet.copyOf(requireNonNull(notNullColumnSymbols, "notNullColumns is null"));
         this.partitioningScheme = requireNonNull(partitioningScheme, "partitioningScheme is null");
+        this.preferredPartitioningScheme = requireNonNull(preferredPartitioningScheme, "preferredPartitioningScheme is null");
```

Comment:
add a validation that just one of `partitioningScheme` and `preferredPartitioningScheme` is set at same time